crates/rolldown/src/types/scan_stage_cache.rs

```diff
@@ -42,10 +42,10 @@ impl ScanStageCache {
     self.snapshot.as_ref().unwrap()
   }
 
-  pub fn merge(&mut self, mut scan_stage_output: ScanStageOutput) {
+  pub fn merge(&mut self, mut scan_stage_output: ScanStageOutput) -> Result<(), &'static str> {
```

Comment:
I don't use a specific `Error` type because currently, there is only one kind of error.

Reply:
You could return build result and use `anyhow::format_err`, which brings error stack trace. I'm not sure if `Err(&'str`  does that,